python/src/filesystem.rs

```diff
@@ -590,31 +580,23 @@ impl ObjectOutputStream {
 
     fn write(&mut self, data: &PyBytes) -> PyResult<i64> {
         self.check_closed()?;
-        let len = data.as_bytes().len() as i64;
         let py = data.py();
-        let data = data.as_bytes();
-        let res = py.allow_threads(|| match rt().block_on(self.writer.write_all(data)) {
-            Ok(_) => Ok(len),
-            Err(err) => {
-                rt().block_on(self.store.abort_multipart(&self.path, &self.multipart_id))
-                    .map_err(PythonError::from)?;
-                Err(PyIOError::new_err(err.to_string()))
-            }
-        })?;
-        self.buffer_size += len;
-        if self.buffer_size >= self.max_buffer_size {
-            let _ = self.flush(py);
-            self.buffer_size = 0;
+        let bytes = data.as_bytes();
+        self.buffer.extend_from_slice(bytes);
+        let len = bytes.len();
+        if self.max_buffer_size >= self.buffer.content_length() {
```

Comment:
This should be the other way around, right?

To only flush when the current buffer is greater than the max_buffer_size (this was added a while ago to prevent deadlocks)

Reply:
Haha yep! Explains that one odd test failure, thank you for catching.